src/test/java/com/google/sps/data/ScheduledInterviewServletTest.java

```diff
@@ -0,0 +1,108 @@
+
+@RunWith(JUnit4.class)
+public final class ScheduledInterviewServletTest {
+  LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalCapabilitiesServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+  // Tests whether a scheduledInterview object was added to datastore.
+  @Test
+  public void validScheduledInterviewServletPostRequest() throws IOException {
+    ScheduledInterviewServlet scheduledInterviewServlet = new ScheduledInterviewServlet();
+    scheduledInterviewServlet.init(new FakeScheduledInterviewDao());
+    MockHttpServletRequest postRequest = new MockHttpServletRequest();
+    MockHttpServletResponse postResponse = new MockHttpServletResponse();
+    postRequest.addParameter("startTime", "2020-07-05T18:30:10Z");
+    postRequest.addParameter("endTime", "2020-07-05T19:30:10Z");
+    postRequest.addParameter("interviewer", "user@company.org");
+    postRequest.addParameter("interviewee", "user@gmail.com");
+
+    scheduledInterviewServlet.doPost(postRequest, postResponse);
+    Assert.assertEquals(200, postResponse.getStatus());
+  }
+
+  // Tests whether a list of scheduledInterviews was returned by the server
+  @Test
+  public void validScheduledInterviewServletGetRequest() throws IOException {
+    ScheduledInterviewServlet scheduledInterviewServlet = new ScheduledInterviewServlet();
+    scheduledInterviewServlet.init(new FakeScheduledInterviewDao());
+    MockHttpServletRequest getRequest = new MockHttpServletRequest();
+    MockHttpServletResponse getResponse = new MockHttpServletResponse();
+    MockHttpServletRequest postRequest = new MockHttpServletRequest();
+
+    postRequest.addParameter("startTime", "2020-07-05T18:30:10Z");
+    postRequest.addParameter("endTime", "2020-07-05T19:30:10Z");
+    postRequest.addParameter("interviewer", "user@company.org");
+    postRequest.addParameter("interviewee", "user@gmail.com");
+    scheduledInterviewServlet.doPost(postRequest, new MockHttpServletResponse());
+
+    postRequest.setParameter("startTime", "2020-07-05T20:30:10Z");
+    postRequest.setParameter("endTime", "2020-07-05T21:30:10Z");
+    postRequest.setParameter("interviewer", "user2@company.org");
+    postRequest.setParameter("interviewee", "user@gmail.com");
+    scheduledInterviewServlet.doPost(postRequest, new MockHttpServletResponse());
+
+    postRequest.setParameter("startTime", "2020-07-05T18:30:10Z");
+    postRequest.setParameter("endTime", "2020-07-05T19:30:10Z");
+    postRequest.setParameter("interviewer", "user2@company.org");
+    postRequest.setParameter("interviewee", "user1@gmail.com");
+    scheduledInterviewServlet.doPost(postRequest, new MockHttpServletResponse());
+
+    getRequest.addParameter("userEmail", "user@gmail.com");
+    scheduledInterviewServlet.doGet(getRequest, getResponse);
+
+    Assert.assertEquals(200, getResponse.getStatus());
```

Comment:
Also make sure getResponse has the right entities in the right order?

Reply:
Done